apps/whale-api/src/module.api/address.controller.ts

```diff
@@ -87,7 +105,7 @@ export class AddressController {
         const start = foundIndex + 1 // plus 1 to exclude the prev txid
         const size = start + query.size
         const sliced = list.slice(start, size)
-        if (sliced.length !== query.size) {
+        if (sliced.length !== query.size && !isLastBatch) {
```

Comment:
`sliced.length !== query.size` is required as https://github.com/JellyfishSDK/jellyfish/pull/1710#issuecomment-1220453064

added `!isLastBatch` checker to stop looping

(took some time on this.. lack of **competitive programming** practice :sweat_smile: )